src/shared/portal.rs

```diff
@@ -0,0 +1,305 @@
+use std::any::Any;
+use std::fmt::Debug;
+
+use makepad_widgets::*;
+
+live_design! {
+    import makepad_widgets::base::*;
+    import makepad_widgets::theme_desktop_dark::*;
+
+    import crate::shared::styles::*;
+
+    PortalView = {{PortalView}} {
+        width: Fill
+        height: Fill
+        flow: Overlay
+    }
+
+    Portal = {{Portal}} {
+        width: Fill
+        height: Fill
+
+        flow: Right
+    }
+}
+
+#[derive(Live, LiveHook, LiveRegisterWidget, WidgetRef)]
+pub struct PortalView {
+    #[deref]
+    view: View,
+}
+
+impl Widget for PortalView {
+    fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
+        self.view.handle_event(cx, event, scope);
+    }
+
+    fn draw_walk(&mut self, cx: &mut Cx2d, scope: &mut Scope, walk: Walk) -> DrawStep {
+        self.view
+            .draw_walk(cx, scope, walk.with_abs_pos(DVec2 { x: 0., y: 0. }))
+    }
+}
+
+impl WidgetNode for PortalView {
+    fn walk(&mut self, cx: &mut Cx) -> Walk {
+        self.view.walk(cx)
+    }
+
+    fn redraw(&mut self, cx: &mut Cx) {
+        self.view.redraw(cx);
+    }
+
+    fn find_widgets(&mut self, path: &[LiveId], cached: WidgetCache, results: &mut WidgetSet) {
+        self.view.find_widgets(path, cached, results);
+    }
+}
+
+impl PortalView {
+    pub fn show(&mut self, cx: &mut Cx) {
+        self.apply_over(cx, live! {visible: true});
+        self.redraw(cx);
+    }
+
+    pub fn hide(&mut self, cx: &mut Cx) {
+        self.apply_over(cx, live! {visible: false});
```

Comment:
Nit: there is a `apply_over_and_redraw` https://github.com/makepad/makepad/blob/rik/widgets/src/widget.rs#L542

Reply:
I tried, but its using this one there: https://github.com/makepad/makepad/blob/rik/platform/src/live_traits.rs#L168